icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -1403,10 +1403,15 @@ private static synchronized void loadAliasData() {
                         "Incorrect key [" + aliasFrom + "] in alias:territory.");
                 }
                 if (aliasTo.length() < 3 || aliasTo.length() > 8) {
-                    // Ignore replacement < 3 for now. see CLDR-14312
-                    // throw new IllegalArgumentException(
-                    //    "Incorrect value [" + aliasTo + "] in alias:subdivision.");
-                    continue;
+                    if (aliasTo.length() == 2) {
```

Comment:
... and if you do want to change the alias data, I would pull the length==2 case out of the error handling for clarity.